spec/services/form_validator_spec.rb

```diff
@@ -2,214 +2,233 @@
+      it "is valid with value" do
+        params.merge!(address1: 'foo')
+        expect(subject).to be_valid
       end
 
-      it "with nil" do
-        params.merge!(address1: nil)
-        expect(subject).to_not be_valid
+      it 'is valid with value and string key' do
+        params.merge!('address1' => 'bar')
+        expect(subject).to be_valid
       end
 
-      it "with false" do
-        params.merge!(address1: false)
-        expect(subject).to_not be_valid
-      end
+      context "is invalid" do
+        it "without value" do
+          expect(subject).to_not be_valid
+        end
 
-      it "with empty string" do
-        params.merge!(address1: "")
-        expect(subject).to_not be_valid
-      end
+        it "with nil" do
+          params.merge!(address1: nil)
+          expect(subject).to_not be_valid
+        end
+
+        it "with false" do
+          params.merge!(address1: false)
+          expect(subject).to_not be_valid
+        end
+
+        it "with empty string" do
+          params.merge!(address1: "")
+          expect(subject).to_not be_valid
+        end
 
-      it "with paragraph data type and empty string" do
-        element.update_attributes(data_type: 'paragraph')
-        params.merge!(address1: "")
-        expect(subject).to_not be_valid
+        it "with paragraph data type and empty string" do
+          element.update_attributes(data_type: 'paragraph')
+          params.merge!(address1: "")
+          expect(subject).to_not be_valid
+        end
       end
     end
-  end
 
-  context "with email as data_type" do
-    let(:element) { create :form_element, :email, form: form }
-    let(:params){ {form_id: element.form_id, email: "foo@example.com"} }
+    context "with email as data_type" do
+      let!(:element) { create :form_element, :email, form: form }
+      let(:params){ {email: "foo@example.com"} }
 
-    context "is valid" do
-      it "with regular address" do
-        expect(subject).to be_valid
+      context "is valid" do
+        it "with regular address" do
+          expect(subject).to be_valid
+        end
+
+        it "with multiple dots" do
+          params.merge!(email: "neal.donnelly@cycles.cs.princeton.edu")
+          expect(subject).to be_valid
+        end
       end
 
-      it "with multiple dots" do
-        params.merge!(email: "neal.donnelly@cycles.cs.princeton.edu")
-        expect(subject).to be_valid
+      context "is invalid" do
+        it "with a basic sentence" do
+          params.merge!(email: "I'm not an email!")
+          expect(subject).to_not be_valid
+        end
+
+        it "with missing the TLD" do
+          params.merge!(email: "neal@sumofus")
+          expect(subject).to_not be_valid
+        end
+
+        it "with two @ symbols" do
+          params.merge!(email: "this@that@other.com")
+          expect(subject).to_not be_valid
+        end
+
+        it 'with ascii' do
+          params.merge!(email: "this\xC2@other.com")
```

Comment:
nah, just badly transcoded strings. also none of this code is new - it just got indented when I put it in a `shared_examples` block